### PR TITLE
logger_delete: map gRPC 'NOT_FOUND' error onto our 'NotFound' exception.

### LINKED ISSUE #2091: Logging system tests:  gRPC/GaxError for 'NotFound' in tearDown / logger.delete
From: https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/151551907#L647-L675

```python
======================================================================
ERROR: test_log_struct_w_metadata (logging_.TestLogging)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/system_tests/logging_.py", line 69, in tearDown
    retry(doomed.delete)()
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/system_tests/retry.py", line 77, in wrapped_function
    return to_wrap(*args, **kwargs)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/gcloud/logging/logger.py", line 268, in delete
    client.logging_api.logger_delete(self.project, self.name)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/gcloud/logging/_gax.py", line 123, in logger_delete
    self._gax_api.delete_log(path, options)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/google/cloud/logging/v2/logging_service_v2_api.py", line 216, in delete_log
    self._delete_log(request, options)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/google/gax/api_callable.py", line 480, in inner
    return api_caller(api_call, this_settings, request)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/google/gax/api_callable.py", line 468, in base_caller
    return api_call(*args)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/google/gax/api_callable.py", line 433, in inner
    raise_with_traceback(GaxError('RPC failed', cause=exception))
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/google/gax/api_callable.py", line 430, in inner
    return a_func(*args, **kwargs)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/google/gax/api_callable.py", line 64, in inner
    return a_func(*updated_args, **kwargs)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/grpc/beta/_client_adaptations.py", line 305, in __call__
    self._request_serializer, self._response_deserializer)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests/lib/python2.7/site-packages/grpc/beta/_client_adaptations.py", line 203, in _blocking_unary_unary
    raise _abortion_error(rpc_error_call)
GaxError: GaxError(RPC failed, caused by AbortionError(code=StatusCode.NOT_FOUND, details="Requested entity was not found."))
```

### PR DESCRIPTION
**Pull request by tseaver**
Fixes #2091.

### CODE REVIEW

File: gcloud/logging/_gax.py [hunk 1]
```diff
@@ -120,7 +120,12 @@ def logger_delete(self, project, logger_name):
         """
         options = None
         path = 'projects/%s/logs/%s' % (project, logger_name)
-        self._gax_api.delete_log(path, options)
+        try:
+            self._gax_api.delete_log(path, options)
+        except GaxError as exc:
+            if exc_to_code(exc.cause) == StatusCode.NOT_FOUND:
+                raise NotFound(path)
```
Comment (dhermes) [inline gcloud/logging/_gax.py]: Could we also provide `NotFound(path, errors=(exc.details,))`?

Comment (dhermes) [inline gcloud/logging/_gax.py]: I based it on

```
GaxError: GaxError(RPC failed, caused by AbortionError(code=StatusCode.NOT_FOUND, details="Requested entity was not found."))
```

Which gives

```
Requested entity was not found.
```

as the details (yes it's in `exc.cause.details`)

Comment (tseaver) [inline gcloud/logging/_gax.py]: In order to stick it into `NotFound._errors`, we would need to synthesize a mapping to hold that text.  Given that it adds no real information, I think we could skip that bit.

Comment (dhermes) [inline gcloud/logging/_gax.py]: What if it becomes useful in the future?

Anyhow, just skip it, that's fine.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**